sky/provision/kubernetes/utils.py

```diff
@@ -891,6 +900,40 @@ def check_credentials(context: Optional[str],
     else:
         return True, None
 
+def _check_pod_config(
+        context: Optional[str] = None, pod_config: Optional[Any] = None) \
+        -> Tuple[bool, Optional[str]]:
+    """Check if the pod_config is a valid pod config
+
+    Using create_namespaced_pod api with dry_run to check the pod_config
+    is valid or not.
+
+    Returns:
+        bool: True if pod_config is valid.
+        str: Error message about why the pod_config is invalid, None otherwise.
+    """
+    try:
+        namespace = get_kube_config_context_namespace(context)
+        kubernetes.core_api(context).create_namespaced_pod(
+            namespace,
+            body=pod_config,
+            dry_run='All',
+            field_validation='Strict',
+            _request_timeout=kubernetes.API_TIMEOUT)
```

Comment:
Does this approach work even if the pod_config is partially specified? E.g., 
```
kubernetes:
  pod_config:
    spec:
      containers:
        - env:
            - name: MY_ENV_VAR
              value: "my_value"
```
My hunch is k8s will reject this pod spec since it's not a complete pod spec, but it's a valid pod_config in our case.

Reply:
yes, the k8s will reject this pod spec. 
if this pod_config is valid in this project. is there any definition about this config?  for example: some filed is required or optional? or all the filed is optional here, but it must follow the k8s pod require only if it has been set ?

Reply:
> all the filed is optional here, but it must follow the k8s pod require only if it has been set ?

Yes, this is the definition of a valid pod_spec. 

>  can check the pod config by using k8s api after combine_pod_config_fields and combine_metadata_fields during launch (that is the early stage of launching.)

Yes, that sounds reasonable as long as we can surface to the user where the error comes in the user's pod config.

Reply:
Have we considered having a simple local schema check, with the json schema fetched and flattened from something like https://github.com/instrumenta/kubernetes-json-schema/tree/master?

Reply:
> Have we considered having a simple local schema check, with the json schema fetched and flattened from something like https://github.com/instrumenta/kubernetes-json-schema/tree/master?

Yeah, I took a look at this before. The main problem with this setup is that it needs to grab JSON schema files from other repo eg: https://github.com/instrumenta/kubernetes-json-schema/tree/master, depending on which version of k8s user using. I'm not sure if it's a good idea for sky to download dependencies to the local machine while it's running. Plus, if we want to check pod_config locally using JSON schema, we might need to let users choose their k8s version so we can get the right schema file.

Reply:
Let's try the approach you proposed above (`check the pod config by using k8s api after combine_pod_config_fields and combine_metadata_fields`) if it can surface the exact errors to the users.

If that does not work, we may need to do schema validation locally. Pod API has been relatively stable, so might not be too bad to have a fixed version schema for validation.

Reply:
LGTM, 
BTW i found a error case when i test the approach with json schema in kubernetes-json-schema.
here is my part of test yaml 
```yaml
containers:
    - name: local_test
       image: test
```
note, the name here `local_test` with `_` inside, it's invalid when we creating a pod, but will pass the check by json schema. 
<img width="1540" alt="image" src="https://github.com/user-attachments/assets/45b54bff-afb9-4ab4-9882-84c1b81e4582" />
and if we use this config to create sky cluster, it will fail later because the invalid name.
<img width="1718" alt="image" src="https://github.com/user-attachments/assets/1de48280-a2f5-484b-9c05-3c1f19c1215b" />